components/omega/OmegaBuild.cmake

```diff
@@ -606,14 +606,24 @@ macro(update_variables)
     set(CMAKE_INSTALL_PREFIX ${OMEGA_INSTALL_PREFIX})
   endif()
 
+  if(NOT DEFINED OMEGA_MPI_ON_DEVICE)
+    option(OMEGA_MPI_ON_DEVICE "Allow device buffers in MPI communication (default ON)." ON)
+  endif()
+
   if("${OMEGA_ARCH}" STREQUAL "CUDA")
     option(Kokkos_ENABLE_CUDA "" ON)
     option(Kokkos_ENABLE_CUDA_LAMBDA "" ON)
     add_definitions(-DOMEGA_TARGET_DEVICE)
+    if(OMEGA_MPI_ON_DEVICE AND OMEGA_CIME_MACHINE STREQUAL "pm-gpu")
+      option(Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC "" OFF)
+    endif()
 
   elseif("${OMEGA_ARCH}" STREQUAL "HIP")
     option(Kokkos_ENABLE_HIP "" ON)
     add_definitions(-DOMEGA_TARGET_DEVICE)
+    if(OMEGA_MPI_ON_DEVICE AND OMEGA_CIME_MACHINE STREQUAL "frontier")
+      file(APPEND ${_EnvScript} "export MPICH_GPU_SUPPORT_ENABLED=1\n\n")
+    endif()
```

Comment:
Because we prefer to enable GPU-aware MPI whenever it is available, we may remove this part and place the following right above the entire if block:

```
if(OMEGA_MPI_ON_DEVICE AND "${MPILIB_NAME}" STREQUAL "mpich")
  file(APPEND ${_EnvScript} "export MPICH_GPU_SUPPORT_ENABLED=1\n\n")
endif()
```
I am not very confident about the impact of setting MPICH_GPU_SUPPORT_ENABLED=1 if the specific MPICH does not support GPU-aware MPI. My guess is that it would simply be ignored, but let me know if that’s not the case.

Reply:
Running with `MPICH_GPU_SUPPORT_ENABLED=1` and `OMEGA_ARCH=SERIAL` on Frontier causes runtime error: 
```
0: MPICH ERROR [Rank 0] [job id 2808765.18] [Tue Nov 26 15:56:34 2024] [frontier00086] - Abort(-1) (rank 0 in comm 0): MPIDI_CRAY_init: GPU_SUPPORT_ENABLED is requested, but GTL library is not linked
```
So I'll probably restrict it so it isn't appended when `OMEGA_ARCH` is `SERIAL` or `OPENMP`